src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/BlobManager.java

```diff
@@ -84,10 +86,22 @@ public CloudBlobContainer getContainer(String containerName) throws URISyntaxExc
     }
 
     public List<CloudBlobContainer> listInputContainers() {
-        return StreamSupport
+        List<CloudBlobContainer> cloudBlobContainerList = StreamSupport
             .stream(cloudBlobClient.listContainers().spliterator(), false)
             .filter(c -> !c.getName().endsWith(REJECTED_CONTAINER_NAME_SUFFIX))
+            .filter(this::filterBySelectedContainer)
             .collect(toList());
+
+        if (cloudBlobContainerList.isEmpty()) {
+            throw new ContainerNotFoundException();
+        }
```

Comment:
hmm. this potentially is "risky" change. Can we return empty list instead and log an error in case not found such container if specified in config?

Reply:
changed

Reply:
this still logs everytime. even if `ALL` is in application config